docs/server-event-types.md

```diff
@@ -0,0 +1,49 @@
+---
+id: server-event-types
+title: Event type reference
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
+- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
+- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
+- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
+- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
+- **EVENT_TYPE_TIMER_STARTED**:
+- **EVENT_TYPE_TIMER_FIRED**:
+- **EVENT_TYPE_TIMER_CANCELED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**:
+- **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_INITIATED**:
+- **EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_EXTERNAL_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
+- **EVENT_TYPE_MARKER_RECORDED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW**:
+- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**:
+- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TERMINATED**:
```

Comment:
- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED**:
This event type indicates that Temporal server will try to start a child workflow.

- **EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the child workflow execution cannot be started / triggered, usually due to child workflow ID collision.

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_STARTED**:
This event type indicates the child workflow execution is successfully started / triggered.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_COMPLETED**:
This event type indicates the child workflow execution is successfully completed.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_FAILED**:
This event type indicates the child workflow execution is unsuccessfully completed.
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_CANCELED**:
This event type indicates the child workflow execution is cancelled. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TIMED_OUT**:
This event type indicates the child workflow execution is timed out by Temporal server. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**

- **EVENT_TYPE_CHILD_WORKFLOW_EXECUTION_TERMINATED**:
This event type indicates the child workflow execution terminated. 
Also see **EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED**